src/API/Management/Users.php

```diff
@@ -84,30 +130,46 @@ public function delete($userId)
      * @param string $userId
      * @param array  $identities
      *
+     * @throws ApiException On invalid responses
+     *
      * @return array
      */
     public function linkAccount($userId, array $identities)
     {
         $response = $this->httpClient->post(sprintf('/users/%s/identities', $userId), [], json_encode($identities));
 
-        return ResponseMediator::getContent($response);
+        if (201 === $response->getStatusCode()) {
+            return ResponseMediator::getContent($response);
+        }
+
+        $this->handleExceptions($response);
     }
 
     /**
+     * @link https://auth0.com/docs/api/management/v2#!/Users/delete_provider_by_user_id
+     *
      * @param string $userId
      * @param string $provider
      * @param string $identityId
      *
+     * @throws ApiException On invalid responses
+     *
      * @return array
      */
     public function unlinkAccount($userId, $provider, $identityId)
     {
         $response = $this->httpClient->delete(sprintf('/users/%s/identities/%s/%s', $userId, $provider, $identityId));
 
-        return ResponseMediator::getContent($response);
+        if (200 === $response->getStatusCode()) {
+            return ResponseMediator::getContent($response);
+        }
+
+        $this->handleExceptions($response);
     }
 
     /**
+     * TODO find a link to the documentation
```

Comment:
the idea was to point to any doc about refresh tokens explaining token revocation. 

This will be deprecated in favor of the new `/oauth/revoke` one I think.